test/e2e/specs/editor/block-computed-styles/paragraph-styles.spec.js

```diff
@@ -0,0 +1,402 @@
+				blocks: {
+					'core/paragraph': styles,
+				},
+			} );
+
+			await admin.createNewPost();
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Hello <a href="https://wordpress.org">world</a>',
+				},
+			} );
+
+			const block = editor.canvas.getByRole( 'document', {
+				name: 'Block: Paragraph',
+			} );
+
+			// fontStyle, fontWeight, fontSize.
+			await expect( block ).toHaveCSS(
+				'font',
+				'200 12px / 14.4px Times'
+			);
+			await expect( block ).toHaveCSS( 'text-transform', 'lowercase' );
+			await expect( block ).toHaveCSS( 'letter-spacing', '1.2px' );
+			await expect( block ).toHaveCSS(
+				'text-decoration',
+				'underline solid rgb(0, 0, 0)'
+			);
+			await expect( block ).toHaveCSS( 'line-height', '14.4px' );
+		} );
+
+		await test.step( 'Block instance styles', async () => {
+			const style = {
+				typography: {
+					fontWeight: '300',
+					textTransform: 'capitalize',
+					textDecoration: 'line-through',
+					letterSpacing: '1.3px',
+					lineHeight: '1.3',
+				},
+			};
+
+			await admin.createNewPost();
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Hello',
+					style,
+					fontSize: '13px',
+				},
+			} );
+
+			const block = editor.canvas.getByRole( 'document', {
+				name: 'Block: Paragraph',
+			} );
+
+			// fontStyle, fontWeight, fontSize.
+			await expect( block ).toHaveCSS(
+				'font',
+				'300 12px / 15.6px Times'
+			);
+			await expect( block ).toHaveCSS( 'text-transform', 'capitalize' );
+			await expect( block ).toHaveCSS( 'letter-spacing', '1.3px' );
+			await expect( block ).toHaveCSS(
+				'text-decoration',
+				'line-through solid rgb(0, 0, 0)'
+			);
+			await expect( block ).toHaveCSS( 'line-height', '15.6px' );
+		} );
+	} );
+
+	test( 'Spacing', async ( { admin, editor, requestUtils } ) => {
+		await test.step( 'Global block styles', async () => {
+			const globalBlockStyles = {
+				spacing: {
+					margin: {
+						top: '1px',
+						bottom: '2px',
+					},
+					padding: {
+						top: '3px',
+						right: '4px',
+						bottom: '5px',
+						left: '6px',
+					},
+				},
+			};
+
+			await requestUtils.updateGlobalStyles( {
+				blocks: {
+					'core/paragraph': globalBlockStyles,
+				},
+			} );
+
+			await admin.createNewPost();
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 1',
+				},
+			} );
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 2',
+				},
+			} );
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 3',
+				},
+			} );
+
+			const paragraphs = editor.canvas.getByRole( 'document', {
+				name: 'Block: Paragraph',
+			} );
+
+			// Left/right margin is set to auto by alignments, there's no reliable way to test this.
+			// First block will have its top margin set to zero by the layout.
```

Comment:
We could wrap the paragraph in a group block and change the group's layout (to `default`) to allow the paragraph's left/right margins to take effect.

Doing this manually in the editor is just toggling off the "Inner blocks use content width" option. For these tests it could just be adding  `layout: { type: 'default' }` to the group block's attribtues.

For example:
<!-- wp:group {"layout":{"type":"default"}} -->

Reply:
Nice idea. I've added extra blocks and assertions for that.